### ISSUE TITLE
[Bug]: Incompatibility with Google Service Location version 21.0.0

### ISSUE THREAD

**Issue by lucasqrib** [Appboy/appboy-android-sdk#247]
### Braze Android SDK Version

23.3.0

### Steps To Reproduce

With the new [version released of the Play Services Location](https://developers.google.com/android/guides/releases#october_13_2022), a break change was introduced: 
- The `FusedLocationProviderClient` was changed to an `Interface` instead of a `Class`

Because of this, updating the project to the new location version (21.0.0), makes the call: 
`  Braze.getInstance(context).requestLocationInitialization()` to fail

### Expected Behavior

The geofencing tracking start with success using the new version of the Google Play Location Services.

### Actual Incorrect Behavior

The initialization failed with and exception

### Verbose Logs

```shell
Child job of SerialCoroutineScope got exception: java.lang.IncompatibleClassChangeError: Found interface com.google.android.gms.location.FusedLocationProviderClient, but class was expected (declaration of 'com.google.android.gms.location.FusedLocationProviderClient' appears in /data/app/com.hearst.android.khtv-f5VRvQKjekDPQ8-OAPEXcA==/base.apk!classes33.dex)
                                                                                                    java.lang.IncompatibleClassChangeError: Found interface com.google.android.gms.location.FusedLocationProviderClient, but class was expected (declaration of 'com.google.android.gms.location.FusedLocationProviderClient' appears in /data/app/com.hearst.android.khtv-f5VRvQKjekDPQ8-OAPEXcA==/base.apk!classes33.dex)
                                                                                                    	at bo.app.r1.a(SourceFile:9)
                                                                                                    	at bo.app.l.a(SourceFile:507)
                                                                                                    	at bo.app.l.b(SourceFile:76)
                                                                                                    	at bo.app.l.b(SourceFile:10)
                                                                                                    	at com.braze.Braze$j2.a(SourceFile:1)
                                                                                                    	at com.braze.Braze$j2.invoke(SourceFile:1)
                                                                                                    	at com.braze.Braze$r2$a.invokeSuspend(SourceFile:2)
                                                                                                    	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
                                                                                                    	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
                                                                                                    	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:284)
                                                                                                    	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
                                                                                                    	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
                                                                                                    	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source:1)
                                                                                                    	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
                                                                                                    	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source:1)
                                                                                                    	at com.braze.Braze$r2.invokeSuspend(SourceFile:2)
                                                                                                    	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
                                                                                                    	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
                                                                                                    	at java.lang.Thread.run(Thread.java:764)
```

### Additional Information

We faced the same issue in other apps, the solution is to update the Google Location library and make some minor changes to comport it.

**Comment by radixdev** [comment]
Hi @lucasqrib ,

We've filed this issue internally and will get back to you with updates/fix.

**Comment by radixdev** [comment]
Hi @lucasqrib ,

A fix was found and will be included in the next release. Thank you for your patience!

**Comment by lucasqrib** [comment]
Many thanks @radixdev !

**Comment by radixdev** [comment]
Hi @lucasqrib ,

This is addressed in version 24.0.0 of the SDK. Please let us know if this fixes your issue!
